src/wsrep_provider_v26.cpp

```diff
@@ -665,6 +668,22 @@ namespace
     }
 #endif /* HAVE_PSI_INTERFACE */
 
+    static wsrep_cb_status enc_get_key_cb(const wsrep_buf_t* keyId,
+                                          wsrep_enc_key_t* key) {
+        std::string skeyId(static_cast<const char*>(keyId->ptr), keyId->len);
+        std::string skey = wsrep_get_master_key(skeyId);
+        if (skey.length() == 0 || skey.length() > key->len) {
+            return WSREP_CB_FAILURE;
+        }
+        memcpy(const_cast<void*>(key->ptr), skey.c_str(), skey.length());
```

Comment:
Isn't a buffer size check missing here? (key->len >= skey.length())

Also, how does the caller know the actual length of the returned key?

Reply:
This key is AES key and needs to be 32 bytes, which is validated by the caller. I've missing handling of key length value returned by `wsrep_get_master_key()` for caller to be able do validation